### PR TITLE
Non default domain

### PR DESCRIPTION
**Pull request by pupapaik**
Adding variable for non default domain together with unit tests.

### CODE REVIEW

**Comment by nati** [comment]
Thank you for your PR!
Code looks good. 
Could you rebase multiple commit for one?
also could you not why we are replacing uuid lib in the commit message?

**Comment by pupapaik** [comment]
Sure, we can do that. Regarding uuid lib, what exactly do you want? because we had to replace libraries, otherwise it does not work.

**Comment by pedro-r-marques** [comment]
@pupapaik Can you please propose the patch first to master ? We can then back port to release-1.1.

File: pkg/network/opencontrail/config.go [hunk 1]
```diff
@@ -30,11 +30,14 @@ import (
 )
 
 const (
-	DefaultDomain        = "default-domain"
+	// DefaultDomain        = "default-domain"
```
Comment (pedro-r-marques) [inline pkg/network/opencontrail/config.go]: Please remove the line.

File: .travis.yml [hunk 2]
```diff
@@ -10,7 +10,10 @@ install:
   - tar -zxvf contrail-go-api-generated-types-r2.20.tar.gz -C ${GOPATH}/src
   - mkdir -p ${GOPATH}/src/k8s.io
   - (cd ${GOPATH}/src/k8s.io && git clone https://github.com/kubernetes/kubernetes.git -b v1.1.1)
-  - go get code.google.com/p/go-uuid/uuid
+  - go get github.com/pborman/uuid
+  # Workaround rename of uuid for dependency components
+  - mkdir -p ${GOPATH}/src/code.google.com/p/go-uuid
```
Comment (pedro-r-marques) [inline .travis.yml]: Please remove this line.

File: .travis.yml [hunk 3]
```diff
@@ -10,7 +10,10 @@ install:
...
+  - git clone https://github.com/pborman/uuid.git ${GOPATH}/src/code.google.com/p/go-uuid/uuid
```
Comment (pedro-r-marques) [inline .travis.yml]: go get should have been sufficient

File: pkg/network/opencontrail/controller_test.go [hunk 4]
```diff
@@ -2112,3 +2112,151 @@ func TestServiceBeforeNamespace(t *testing.T) {
 	_, err := types.VirtualNetworkByName(client, "default-domain:newns:service-foo")
 	assert.NoError(t, err)
 }
+
+
+func TestDomainVariable(t *testing.T) {
+	kube := mocks.NewKubeClient()
+
+	client := new(contrail_mocks.ApiClient)
```
Comment (pedro-r-marques) [inline pkg/network/opencontrail/controller_test.go]: Use createTestClient()

File: pkg/network/opencontrail/controller_test.go [hunk 5]
```diff
@@ -2112,3 +2112,151 @@ func TestServiceBeforeNamespace(t *testing.T) {
...
+	client.Init()
+
+	client.AddInterceptor("virtual-machine-interface", &VmiInterceptor{})
+	client.AddInterceptor("virtual-network", &NetworkInterceptor{})
+
+	allocator := new(mocks.AddressAllocator)
+
+	controller := NewTestController(kube, client, allocator, nil)
```
Comment (pedro-r-marques) [inline pkg/network/opencontrail/controller_test.go]: pass a nil allocator unless you want to override the allocator for some reasons.

File: pkg/network/opencontrail/controller_test.go [hunk 6]
```diff
@@ -2112,3 +2112,151 @@ func TestServiceBeforeNamespace(t *testing.T) {
...
+			ClusterIP: "10.254.42.42",
+		},
+	}
+
+	glog.Infof("Domain name is %s", controller.config.DefaultDomain)
+	netnsProject := new(types.Project)
+	netnsProject.SetFQName("", []string{controller.config.DefaultDomain, "testns"})
+	client.Create(netnsProject)
+
+	allocator.On("LocateIpAddress", string(pod1.ObjectMeta.UID)).Return("10.0.0.1", nil)
```
Comment (pedro-r-marques) [inline pkg/network/opencontrail/controller_test.go]: You don't need to mock the allocator calls if you let NewTestController create a real allocator.

**Comment by pedro-r-marques** [comment]
@pupapaik The master branch is already using the new uuid library just follow that.

**Comment by pupapaik** [comment]
Cherry-pick from master branch, but I removed UpdateStatus method from /pkg/network/opencontrail/mocks/KubeClient.go, which was related to kubernetes master branch.

**Pull request closed without merging**